ir/src/lib.rs

```diff
@@ -179,21 +179,20 @@ impl AirIR {
         self.boundary_constraints.aux_last()
     }
 
-    // --- PUBLIC ACCESSORS FOR TRANSITION CONSTRAINTS --------------------------------------------
+    // --- PUBLIC ACCESSORS FOR INTEGRITY CONSTRAINTS --------------------------------------------
 
     pub fn constraint_degrees(
         &self,
         trace_segment: TraceSegment,
     ) -> Vec<TransitionConstraintDegree> {
-        self.transition_constraints
-            .constraint_degrees(trace_segment)
+        self.integrity_constraints.constraint_degrees(trace_segment)
     }
 
-    pub fn transition_constraints(&self, trace_segment: TraceSegment) -> &[NodeIndex] {
-        self.transition_constraints.constraints(trace_segment)
+    pub fn integrity_constraints(&self, trace_segment: TraceSegment) -> &[NodeIndex] {
+        self.integrity_constraints.constraints(trace_segment)
     }
 
-    pub fn transition_graph(&self) -> &AlgebraicGraph {
-        self.transition_constraints.graph()
+    pub fn integrity_graph(&self) -> &AlgebraicGraph {
+        self.integrity_constraints.graph()
```

Comment:
I'm not sure about the name `integrity_graph`. The rest of the renaming mostly makes sense and preserves meaning, but this feels like the meaning changes a bit. I think it might be better to call this `constraint_graph`. What do you think? I know it doesn't include the boundary constraints, but I think that's ok (and at some point in the future it might).

Reply:
I would do `constraint_graph`

Reply:
I like `constraint_graph`. I actually think that in the future we'll probably fold boundary constraints into this graph as well.